### PR TITLE
New resource: aws_apigatewayv2_api

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Relates https://github.com/terraform-providers/terraform-provider-aws/issues/7004.

I included the commits from @mbonig - https://github.com/mbonig/terraform-provider-aws/tree/f-gatewayv2, removed the `aws_api_gateway_v2_route` resource to add in a subsequent PR, added documentation, tidied up the code and got acceptance tests to pass.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
FEATURES:

* New Resource: aws_apigatewayv2_api
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSAPIGateway2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Api_basic
=== PAUSE TestAccAWSAPIGateway2Api_basic
=== RUN   TestAccAWSAPIGateway2Api_AllAttributes
=== PAUSE TestAccAWSAPIGateway2Api_AllAttributes
=== RUN   TestAccAWSAPIGateway2Api_Tags
=== PAUSE TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_basic
=== CONT  TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_AllAttributes
--- PASS: TestAccAWSAPIGateway2Api_basic (23.00s)
--- PASS: TestAccAWSAPIGateway2Api_Tags (37.32s)
--- PASS: TestAccAWSAPIGateway2Api_AllAttributes (65.17s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	65.240s
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSAPIGateway2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Api_basic
=== PAUSE TestAccAWSAPIGateway2Api_basic
=== RUN   TestAccAWSAPIGateway2Api_update
=== PAUSE TestAccAWSAPIGateway2Api_update
=== CONT  TestAccAWSAPIGateway2Api_basic
=== CONT  TestAccAWSAPIGateway2Api_update
--- PASS: TestAccAWSAPIGateway2Api_basic (23.43s)
--- PASS: TestAccAWSAPIGateway2Api_update (66.44s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	66.506s
```

**Comment by ewbankkit** [comment]
Rebased to add support for tags enabled via #9153.

**Comment by ewbankkit** [comment]
Test sweeper:

```console
$ SWEEP=us-west-2 SWEEPARGS=-sweep-run=aws_api_gateway_v2_api make sweep
WARNING: This will destroy infrastructure. Use only in development accounts.
go test ./... -v -sweep=us-west-2 -sweep-run=aws_api_gateway_v2_api
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
2019/07/12 15:43:02 [DEBUG] Running Sweepers for region (us-west-2):
2019/07/12 15:43:02 [INFO] Building AWS auth structure
2019/07/12 15:43:02 [INFO] Setting AWS metadata API timeout to 100ms
2019/07/12 15:43:03 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2019/07/12 15:43:03 [INFO] AWS Auth provider used: "EnvProvider"
2019/07/12 15:43:03 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2019/07/12 15:43:03 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2019/07/12 15:43:04 [INFO] Deleting API Gateway v2 API: 0xdnxeagx4
2019/07/12 15:43:06 [INFO] Deleting API Gateway v2 API: 1euwhap7li
2019/07/12 15:43:06 [INFO] Deleting API Gateway v2 API: 46tnfzqtjg
2019/07/12 15:43:07 [INFO] Deleting API Gateway v2 API: 5qrmhx3ls8
2019/07/12 15:43:08 [INFO] Deleting API Gateway v2 API: 6um3maylxc
2019/07/12 15:43:09 [INFO] Deleting API Gateway v2 API: 7lrwzvg1l5
2019/07/12 15:43:10 [INFO] Deleting API Gateway v2 API: a944fj0y4c
2019/07/12 15:43:11 [INFO] Deleting API Gateway v2 API: dz58flf4f4
2019/07/12 15:43:12 Sweeper Tests ran:
	- aws_api_gateway_v2_api
ok  	github.com/terraform-providers/terraform-provider-aws/aws	9.739s
```

**Comment by ewbankkit** [comment]
Rebased to fix merge conflict.

**Comment by ewbankkit** [comment]
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSAPIGateway2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Api_basic
=== PAUSE TestAccAWSAPIGateway2Api_basic
=== RUN   TestAccAWSAPIGateway2Api_AllAttributes
=== PAUSE TestAccAWSAPIGateway2Api_AllAttributes
=== RUN   TestAccAWSAPIGateway2Api_Tags
=== PAUSE TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_basic
=== CONT  TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_AllAttributes
--- PASS: TestAccAWSAPIGateway2Api_basic (19.53s)
--- PASS: TestAccAWSAPIGateway2Api_Tags (30.86s)
--- PASS: TestAccAWSAPIGateway2Api_AllAttributes (52.58s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	52.611s
```

**Comment by ewbankkit** [comment]
### TODO

Use new `internal/keyvaluetags` functionality: https://github.com/terraform-providers/terraform-provider-aws/pull/10018.

**Comment by ewbankkit** [comment]
Rebased and did the Terraform Plugin SDK migration for the new resources.
Use `internal/keyvaluetags` for tagging - https://github.com/terraform-providers/terraform-provider-aws/pull/10018.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGateway2Api_ -timeout 120m
go: finding github.com/terraform-providers/terraform-provider-tls v2.1.1+incompatible
go: finding github.com/terraform-providers/terraform-provider-tls v2.1.1+incompatible
=== RUN   TestAccAWSAPIGateway2Api_basic
=== PAUSE TestAccAWSAPIGateway2Api_basic
=== RUN   TestAccAWSAPIGateway2Api_AllAttributes
=== PAUSE TestAccAWSAPIGateway2Api_AllAttributes
=== RUN   TestAccAWSAPIGateway2Api_Tags
=== PAUSE TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_basic
=== CONT  TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_AllAttributes
--- PASS: TestAccAWSAPIGateway2Api_basic (30.13s)
--- PASS: TestAccAWSAPIGateway2Api_Tags (30.68s)
--- PASS: TestAccAWSAPIGateway2Api_AllAttributes (51.21s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	51.230s
```

**Comment by mbonig** [comment]
Why is this taking so long to get merged in? Are there concerns from the maintainers? Or is this just another issue/pr that's behind thousands of others?

File: aws/provider_test.go [hunk 1]
```diff
@@ -171,6 +171,41 @@ func testAccMatchResourceAttrGlobalARN(resourceName, attributeName, arnService s
...
+			Resource:  arnResource,
+			Service:   arnService,
+		}.String()
+		return resource.TestCheckResourceAttr(resourceName, attributeName, attributeValue)(s)
+	}
+}
+*/
+
+// testAccMatchResourceAttrAnonymousRegionalARN ensures the Terraform state regexp matches a formatted ARN with region but without an account ID
+func testAccMatchResourceAttrAnonymousRegionalARN(resourceName, attributeName, arnService string, arnResourceRegexp *regexp.Regexp) resource.TestCheckFunc {
```
Comment (ewbankkit) [inline aws/provider_test.go]: Track https://github.com/terraform-providers/terraform-provider-aws/pull/10581 which introduces `testAccMatchResourceAttrRegionalARNNoAccount()`.

Comment (ewbankkit) [reply aws/provider_test.go]: Track https://github.com/terraform-providers/terraform-provider-aws/pull/10746 which introduces `testAccCheckResourceAttrGlobalARNNoAccount()`.

**Comment by ewbankkit** [comment]
### TODO
Remove`testAccMatchResourceAttrAnonymousRegionalARN()` and use `testAccMatchResourceAttrRegionalARNNoAccount()`.

**Comment by ewbankkit** [comment]
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGateway2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Api_basic
=== PAUSE TestAccAWSAPIGateway2Api_basic
=== RUN   TestAccAWSAPIGateway2Api_AllAttributes
=== PAUSE TestAccAWSAPIGateway2Api_AllAttributes
=== RUN   TestAccAWSAPIGateway2Api_Tags
=== PAUSE TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_basic
=== CONT  TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_AllAttributes
--- PASS: TestAccAWSAPIGateway2Api_basic (23.26s)
--- PASS: TestAccAWSAPIGateway2Api_Tags (35.73s)
--- PASS: TestAccAWSAPIGateway2Api_AllAttributes (63.20s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	63.276s
```

**Comment by ewbankkit** [comment]
### TODO

* Any documentation changes for https://github.com/terraform-providers/terraform-provider-aws/pull/10684

File: aws/resource_aws_api_gateway2_api.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/apigatewayv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsApiGateway2Api() *schema.Resource {
```
Comment (kreempuff) [inline aws/resource_aws_api_gateway2_api.go]:
```suggestion
func resourceAwsApiGatewayV2Api() *schema.Resource {
```
Should this and the file itself contain `V2` (i.e. `resource_aws_api_gatewayv2_api`) to keep consistent with AWS naming?

Comment (ewbankkit) [reply aws/resource_aws_api_gateway2_api.go]: @kreempuff I followed the naming for CloudHSM V2 - for example [`aws_cloudhsm_v2_cluster`](https://www.terraform.io/docs/providers/aws/r/cloudhsm_v2_cluster.html) resource is implemented in file [`resource_aws_cloudhsm2_cluster.go`](https://github.com/terraform-providers/terraform-provider-aws/blob/34ff52a5834641323eb566ae731f36e6814bc268/aws/resource_aws_cloudhsm2_cluster.go) with entrypoint [`resourceAwsCloudHsm2Cluster()`](https://github.com/terraform-providers/terraform-provider-aws/blob/34ff52a5834641323eb566ae731f36e6814bc268/aws/resource_aws_cloudhsm2_cluster.go#L16).
Happy to change.

Comment (kreempuff) [reply aws/resource_aws_api_gateway2_api.go]: @ewbankkit Makes sense but `CloudHSM`looks like it replaced the previous HSM service as an upgrade, whereas API Gateway and API Gateway V2 exist side by side as two similar but different services. 

I don't think it needs to be changed, but it's a little food for thought 👍 


Great job either way. Can't wait for this to be merged

**Comment by ewbankkit** [comment]
#### TODO

With [AWS SDK v1.25.48](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.48) it looks like we can now tag APIs on creation.

My initial thoughts are **NOT** to add the changes for [HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api.html) in this PR but to get this (and as many of the cascading `aws_api_gateway_v2_*` PRs as possible, which will require some rebasing) merged and then do HTTP API changes as a separate set of PRs.

**Comment by marcoreni** [comment]
Is there any way we can get this (and related apigwv2 resources) merged? That way we could work on expanding this to support HTTP APIs.

@bflad

**Comment by gaw508** [comment]
Any update on this being merged? Any reason why it hasn't been so far? I'm happy to get involved and help if that is the case.

**Comment by smd1000** [comment]
bump

**Comment by ewbankkit** [comment]
Rebased to remove merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGateway2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Api_basic
=== PAUSE TestAccAWSAPIGateway2Api_basic
=== RUN   TestAccAWSAPIGateway2Api_AllAttributes
=== PAUSE TestAccAWSAPIGateway2Api_AllAttributes
=== RUN   TestAccAWSAPIGateway2Api_Tags
=== PAUSE TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_basic
=== CONT  TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_AllAttributes
--- PASS: TestAccAWSAPIGateway2Api_basic (25.03s)
--- PASS: TestAccAWSAPIGateway2Api_Tags (39.52s)
--- PASS: TestAccAWSAPIGateway2Api_AllAttributes (71.28s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	71.318s
```

**Comment by ewbankkit** [comment]
#### TODO
* Fix Travis errors
* Add an `_disappears` acceptance test

**Comment by ewbankkit** [comment]
Re-ran acceptance tests:

```console
% make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGateway2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGateway2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGateway2Api_basic
=== PAUSE TestAccAWSAPIGateway2Api_basic
=== RUN   TestAccAWSAPIGateway2Api_disappears
=== PAUSE TestAccAWSAPIGateway2Api_disappears
=== RUN   TestAccAWSAPIGateway2Api_AllAttributes
=== PAUSE TestAccAWSAPIGateway2Api_AllAttributes
=== RUN   TestAccAWSAPIGateway2Api_Tags
=== PAUSE TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_basic
=== CONT  TestAccAWSAPIGateway2Api_Tags
=== CONT  TestAccAWSAPIGateway2Api_AllAttributes
=== CONT  TestAccAWSAPIGateway2Api_disappears
--- PASS: TestAccAWSAPIGateway2Api_disappears (22.29s)
--- PASS: TestAccAWSAPIGateway2Api_basic (29.34s)
--- PASS: TestAccAWSAPIGateway2Api_Tags (50.84s)
--- PASS: TestAccAWSAPIGateway2Api_AllAttributes (85.04s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	86.330s
```

**Comment by ewbankkit** [comment]
Rebased to remove merge conflicts.

**Comment by ewbankkit** [comment]
Cleaned up function names similar to https://github.com/terraform-providers/terraform-provider-aws/pull/12299 (@kreempuff was correct).
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_basic
=== PAUSE TestAccAWSAPIGatewayV2Api_basic
=== RUN   TestAccAWSAPIGatewayV2Api_disappears
=== PAUSE TestAccAWSAPIGatewayV2Api_disappears
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributes
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributes
=== RUN   TestAccAWSAPIGatewayV2Api_Tags
=== PAUSE TestAccAWSAPIGatewayV2Api_Tags
=== CONT  TestAccAWSAPIGatewayV2Api_basic
=== CONT  TestAccAWSAPIGatewayV2Api_Tags
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributes
=== CONT  TestAccAWSAPIGatewayV2Api_disappears
--- PASS: TestAccAWSAPIGatewayV2Api_disappears (18.93s)
--- PASS: TestAccAWSAPIGatewayV2Api_basic (25.30s)
--- PASS: TestAccAWSAPIGatewayV2Api_Tags (41.03s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributes (84.80s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	84.865s
```

File: aws/provider.go [hunk 3]
```diff
@@ -354,6 +354,7 @@ func Provider() terraform.ResourceProvider {
 			"aws_api_gateway_usage_plan":                              resourceAwsApiGatewayUsagePlan(),
 			"aws_api_gateway_usage_plan_key":                          resourceAwsApiGatewayUsagePlanKey(),
 			"aws_api_gateway_vpc_link":                                resourceAwsApiGatewayVpcLink(),
+			"aws_api_gateway_v2_api":                                  resourceAwsApiGatewayV2Api(),
```
Comment (gdavison) [inline aws/provider.go]: Based on our naming standards at https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#new-service, the service name should be`apigatewayv2`, and therefore the resource should be `aws_apigatewayv2_api`.

Comment (ewbankkit) [reply aws/provider.go]: Sure. For completeness I'll rename the files to `resource_aws_api_gatewayv2_api.go`, `resource_aws_api_gatewayv2_api_test.go` and `api_gatewayv2_api.html.markdown`.

File: aws/resource_aws_api_gateway2_api.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"protocol_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					apigatewayv2.ProtocolTypeWebsocket,
```
[the remark this answers is not in the document]
Comment (ewbankkit) [reply aws/resource_aws_api_gateway2_api.go]: Done.

File: website/allowed-subcategories.txt [hunk 5]
```diff
@@ -2,6 +2,7 @@
 ACM PCA
 ACM
 API Gateway
+API Gateway v2
```
[the remark this answers is not in the document]
Comment (ewbankkit) [reply website/allowed-subcategories.txt]: Done.

File: website/docs/r/api_gateway_rest_api.html.markdown [hunk 6]
```diff
@@ -10,6 +10,8 @@ description: |-
 
 Provides an API Gateway REST API.
 
+-> **Note:** Amazon API Gateway Version 1 resources are used for creating and deploying REST APIs. To create and deploy WebSocket APIs, use Amazon API Gateway Version 2 [resources](https://www.terraform.io/docs/providers/aws/r/api_gateway_v2_api.html).
```
[the remark this answers is not in the document]
Comment (ewbankkit) [reply website/docs/r/api_gateway_rest_api.html.markdown]: Done.

File: aws/resource_aws_api_gateway2_api_test.go [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+		for _, api := range output.Items {
+			log.Printf("[INFO] Deleting API Gateway v2 API: %s", aws.StringValue(api.ApiId))
+			_, err := conn.DeleteApi(&apigatewayv2.DeleteApiInput{
+				ApiId: api.ApiId,
+			})
+			if isAWSErr(err, apigatewayv2.ErrCodeNotFoundException, "") {
+				continue
+			}
+			if err != nil {
+				return fmt.Errorf("error deleting API Gateway v2 API (%s): %s", aws.StringValue(api.ApiId), err)
```
[the remark this answers is not in the document]
Comment (ewbankkit) [reply aws/resource_aws_api_gateway2_api_test.go]: Done.

File: aws/provider.go (hunk 3 above)
Comment (ewbankkit) [reply aws/provider.go]: Done.

**Comment by ewbankkit** [comment]
@gdavison I made all the suggested changes.
For HTTP API support, I enable `protocol_type = "HTTP"` (and added some additional acceptance tests) but did not add any of the new attributes that are applicable only to HTTP APIs. They can be added as part of implementing https://github.com/terraform-providers/terraform-provider-aws/issues/11148.
I will make equivalent changes to the subsequent new API Gateway v2 resources.

File: aws/provider.go (hunk 3 above)
Comment (gdavison) [reply aws/provider.go]: One more tweak, please: can you change it to `aws_apigatewayv2_api`, without the underscore in `apigateway`? We know it's not consistent with v1, but we're trying to standardize on our service naming.

Comment (ewbankkit) [reply aws/provider.go]: Too close to the issue to see the additional `_` 😄.
I have renamed the resource to `aws_apigatewayv2_api`.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_basicWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Api_basicWebSocket
=== RUN   TestAccAWSAPIGatewayV2Api_basicHttp
=== PAUSE TestAccAWSAPIGatewayV2Api_basicHttp
=== RUN   TestAccAWSAPIGatewayV2Api_disappears
=== PAUSE TestAccAWSAPIGatewayV2Api_disappears
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== RUN   TestAccAWSAPIGatewayV2Api_Tags
=== PAUSE TestAccAWSAPIGatewayV2Api_Tags
=== CONT  TestAccAWSAPIGatewayV2Api_basicWebSocket
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== CONT  TestAccAWSAPIGatewayV2Api_Tags
=== CONT  TestAccAWSAPIGatewayV2Api_disappears
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== CONT  TestAccAWSAPIGatewayV2Api_basicHttp
--- PASS: TestAccAWSAPIGatewayV2Api_disappears (21.75s)
--- PASS: TestAccAWSAPIGatewayV2Api_basicWebSocket (26.20s)
--- PASS: TestAccAWSAPIGatewayV2Api_basicHttp (38.05s)
--- PASS: TestAccAWSAPIGatewayV2Api_Tags (44.29s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributesHttp (74.49s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket (75.41s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	75.468s
```

**Pull request merged**